Content.Shared/Health/BodySystem/Surgery/GenericSurgeryUIMessages.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components.UserInterface;
+using Robust.Shared.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Content.Shared.BodySystem
+{
+
+    /// <summary>
+    ///     Used to determine which callback is used by the server after an option is selected.
+    /// </summary>	   
+    public enum SurgeryUIMessageType { SelectBodyPart, SelectMechanism, SelectBodyPartSlot }
```

Comment:
Any particular reason you didn't just have multiple, distinct, message types? You're basically just re-inventing a worse type system.

If they're all going to have the `Targets` dict, then just have them subclass an abstract `UpdateSurgeryUIMessage`,
if they're NOT all going to share that dict, then it's a bonus and you can avoid allocating space for the dict in the messsage.

```csharp
public abstract class UpdateSurgeryUIMessage : BoundUserInterfaceMessage
{
    public Dictionary<string, object> Targets;
    public UpdateSurgeryUIMessage(Dictionary<string, object> targets)
    {
        Targets = targets;
    }
}

public class SelectBodyPartSurgeryUIMessage : UpdateSurgeryUIMessage {};
public class SelectMechanismSurgeryUIMessage : UpdateSurgeryUIMessage {};
public class SelectBodyPartSlotSurgeryUIMessage : UpdateSurgeryUIMessage {};
```

Reply:
I did it since I would have to define a bunch of different messages, but you're right, so I've changed them to be separate messages with callbacks instead of the enum.